### PR TITLE
Update to mozjs 84.

### PR DESCRIPTION
**Pull request by jdm**
Depends on https://github.com/servo/mozjs/pull/265.

### CODE REVIEW

**Review by emilio — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -58,4 +58,9 @@ lazy_static = "1"
 libc = "0.2"
 log = "0.4"
 num-traits = "0.2"
-mozjs_sys = { git = "https://github.com/servo/mozjs", rev = "26a1e8afdb21beec33373ef2a38131272d064bdd" }
+#mozjs_sys = { git = "https://github.com/servo/mozjs", rev = "26a1e8afdb21beec33373ef2a38131272d064bdd" }
+
+mozjs_sys = { git = "https://github.com/servo/mozjs", branch = "smuppy2" }
```
Comment (asajeffrey) [inline Cargo.toml:63]: To be removed once https://github.com/servo/mozjs/pull/265 lands?

File: Cargo.toml [hunk 2]
```diff
@@ -58,4 +58,9 @@ lazy_static = "1"
...
+
+#[patch."https://github.com/servo/mozjs"]
+#mozjs_sys = { path = "../mozjs" }
```
Comment (asajeffrey) [inline Cargo.toml:66]: Ditto?

**Review by asajeffrey — APPROVED** [review]
More yay!

**Pull request merged**